datafusion/common/src/scalar/mod.rs

```diff
@@ -15,7 +15,9 @@
 // specific language governing permissions and limitations
 // under the License.
 
-//! This module provides ScalarValue, an enum that can be used for storage of single elements
+//! [`ScalarValue`]: stores single constant values
```

Comment:
constant may be misleading. 

Maybe like 
```
In databases, a scalar value is simply a single value, as opposed to a set of values or a combination of values. It's like having just one piece of information
```

Reply:
Scalar value is a single value of different datatypes: bool, ints, floats, arrays, structs, etc

Reply:
Good point -- reworded